### PR TITLE
Implement code style rules

### PR DESCRIPTION
**Pull request by martinkearn**
This PR configured the projects to use code analysis as per: https://github.com/buerokratt/Cross-functional-requirements/blob/main/docs/technical/code-analysis.md

The PR also fixes rule violations.

The PR also updates the CI pipeline so that it correctly uses `dotnet format`.

### CODE REVIEW

**Review by joonasroosalung — APPROVED** [review]

File: src/MockClassifier.Api/Controllers/InstitutionController.cs [hunk 1]
```diff
@@ -28,20 +28,25 @@ public InstitutionController(IDmrService dmrService, ITokenService tokenService,
         [HttpPost]
         public AcceptedResult Post([FromBody] MessagesInput messages)
         {
-            foreach( var message in messages.Messages)
+            if (messages == null)
+            {
+                throw new ArgumentNullException(nameof(messages));
```
Comment (yahya130) [inline src/MockClassifier.Api/Controllers/InstitutionController.cs:33]: Would we want to add more relevant HTTP error handling so the calling client receives errors in HTTP context e.g. 
```suggestion
        var resp = new HttpResponseMessage(HttpStatusCode.BadRequest)
        {
            ReasonPhrase = "Message input empty"
        };
        throw new HttpResponseException(resp);
```

File: src/MockClassifier.Api/Controllers/InstitutionController.cs [hunk 2]
```diff
@@ -28,20 +28,25 @@ public InstitutionController(IDmrService dmrService, ITokenService tokenService,
...
                 {
-                     _dmrService.RecordRequest(GetDmrRequest(message, ministry, messages.CallbackUri));
+                    _dmrService.RecordRequest(GetDmrRequest(message, ministry, messages.CallbackUri));
                 }
             }
             return Accepted();
         }
 
-        private static DmrRequest GetDmrRequest(string message, string ministry, string callbackUri)
+        private static DmrRequest GetDmrRequest(string message, string ministry, Uri callbackUri)
```
Comment (yahya130) [inline src/MockClassifier.Api/Controllers/InstitutionController.cs:49]: Would be nice to move this logic into another class, perhaps the DmrRequest class? would be nicer design if the controller only contained API route methods

File: src/MockClassifier.Api/Models/MessagesInput.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@ namespace MockClassifier.Api.Models
     [ExcludeFromCodeCoverage]
     public class MessagesInput
     {
-        public string CallbackUri { get; set; }
-        public string[] Messages { get; set; }
+        public Uri CallbackUri { get; set; }
```
Comment (yahya130) [inline src/MockClassifier.Api/Models/MessagesInput.cs:9]: Good spot :)

File: src/MockClassifier.Api/Controllers/InstitutionController.cs [hunk 4]
```diff
@@ -28,20 +28,25 @@ public InstitutionController(IDmrService dmrService, ITokenService tokenService,
...
-            foreach( var message in messages.Messages)
+            if (messages == null)
+            {
+                throw new ArgumentNullException(nameof(messages));
+            }
+
+            foreach (var message in messages.Messages)
             {
                 List<string> ministries = _naturalLanguageService.Classify(message).ToList();
                 ministries = ministries.Concat(_tokenService.Classify(message).ToList()).ToList();
```
[the remark this answers is not in the document]
Comment (martinkearn) [reply src/MockClassifier.Api/Controllers/InstitutionController.cs:39]: I agree in principle, but this code is not new in the PR so I'm reluctant to change it without fully evaluating the implications. This PR is just about making the minimum changes to the code to address the code analysis violations, so I'll leave this I think.

File: src/MockClassifier.Api/Models/MessagesInput.cs (hunk 3 above)
Comment (martinkearn) [reply src/MockClassifier.Api/Models/MessagesInput.cs:9]: The code analysis forced me into it! :)

File: src/MockClassifier.Api/Controllers/InstitutionController.cs (hunk 2 above)
Comment (martinkearn) [reply src/MockClassifier.Api/Controllers/InstitutionController.cs:49]: See other responses. Refactoring to make the code better reaches beyond the scope of this PR and I don't want to make changes like this without fully evaluating the implications.

**Pull request merged**